source/util/Retry.h

```diff
@@ -61,11 +57,14 @@ namespace Aws
                      * @param retryableFunction the function to retry. This function should return a bool indicating
                      * whether it is successful or not, since this indicator is what will determine whether the
                      * function is retried or not.
+                     * @param onComplete a callback function which will be executed once this function is finished
+                     * attempting retries
                      * @param config the ExponentialRetryConfig specifying whether the function should be retried
                      * @return a bool representing whether the retryableFunction was successful or not
                      */
                     static bool exponentialBackoff(
                         std::function<bool()> retryableFunction,
+                        std::function<void()> onComplete,
```

Comment:
Would it be worth adding this function ptr to the ExponentialRetryConfig instead of passing it in this way?  That way we can have nullptr default by default and a user doesn't need to think about passing something in.

Could also change that function ->static bool exponentialBackoff(std::function<bool()> retryableFunction, ExponentialRetryConfig config, std::function<void()> onComplete
 = nullptr); making it optional

Reply:
I think I like your second approach better, as it will maintain some consistency with the method signatures (IE I won't have one callback function as a parameter to the function and another as a parameter to the config struct.